cogboard-webapp/src/components/widgets/dialogFields/JiraBucketsInput.js

```diff
@@ -0,0 +1,184 @@
+import React, { useState } from 'react';
+import {
+  FormControl,
+  IconButton,
+  List,
+  ListItem,
+  ListItemSecondaryAction,
+  ListItemText
+} from '@material-ui/core';
+import DeleteIcon from '@material-ui/icons/Delete';
+import EditIcon from '@material-ui/icons/Edit';
+import CheckIcon from '@material-ui/icons/Check';
+import { remove } from 'ramda';
+import { v4 } from 'uuid';
+import { prepareChangeEvent } from './helpers';
+import AddIcon from '@material-ui/icons/Add';
+import Input from '@material-ui/core/Input';
+import Fab from '@material-ui/core/Fab';
+import styled from '@emotion/styled/macro';
+
+const StyledInput = styled(Input)`
+  margin-top: 16px;
+  margin-bottom: 8px;
+`;
+const StyledFab = styled(Fab)`
+  margin-top: 16px;
+  margin-bottom: 8px;
+`;
+const StyledList = styled(List)`
+  margin-top: 16px;
+`;
+const JiraBucketsInput = ({ value, onChange }) => {
+  const [buckets, setBuckets] = useState(() =>
+    (value || []).map(bucket => {
+      return {
+        id: v4(),
+        bucketName: bucket.bucketName,
+        jqlQuery: bucket.jqlQuery
+      };
+    })
+  );
+  const [formValueJqlQuery, setFormValueJqlQuery] = useState('');
+  const [formValueBucketName, setFormValueBucketName] = useState('');
+  const [editMode, setEditMode] = useState(false);
+  const handleChangeValJqlQuery = event =>
+    setFormValueJqlQuery(event.target.value);
+  const handleChangeValBucketName = event =>
+    setFormValueBucketName(event.target.value);
```

Comment:
Move those oneliners to the top of this body.

Reply:
Done